tests/Task.test.ts

```diff
@@ -394,9 +394,27 @@ describe('toggle done', () => {
         expect(toggled).not.toBeNull();
         expect(toggled!.status).toStrictEqual(Status.Done);
         expect(toggled!.doneDate).not.toBeNull();
+        expect(toggled!.originalStatusCharacter).toStrictEqual('x');
         expect(toggled!.blockLink).toEqual(' ^my-precious');
     });
 
+    it('removes done date after untoggle', () => {
+        // Arrange
+        const line = '- [x] I thought I finished ✅ 2021-09-12';
+
+        // Act
+        const task: Task = fromLine({
+            line,
+        }) as Task;
+        const toggled: Task = task.toggle()[0];
```

Comment:
This is absolutely fine, and doesn't need changing, but in case it helps you in future, I thought I would point out one small thing here.

The `[0]` makes an assumption that there is only one task returned. Which ought to be the case, because this is a non-recurring task. And if the recurrence code got broken so badly that this returned more than 1 task, other tests would find it.

But if you ever wanted to make a test like this more bullet-proof, you could save the array/list returned from toggle, expect that it has 1 value, and then continue with the rest of the tests.

When reviewing my own tests, I try to always think about ways that a test could accidentally not be testing the thing that I think it is testing...